cluster/state/cluster.go

```diff
@@ -108,175 +107,17 @@ func (c Cluster) NodeIdx(pID peer.ID) (cluster.NodeIdx, error) {
-	return to0xHex(v.PubKey)
-}
-
-// PublicShare returns a peer's threshold BLS public share.
-func (v Validator) PublicShare(peerIdx int) (tbls.PublicKey, error) {
-	return tblsconv.PubkeyFromBytes(v.PubShares[peerIdx])
-}
-
-// toSSZ returns a SSZ friendly version of the validator.
-func (v Validator) toSSZ() validatorSSZ {
-	var pubshares []sszPubkey
-	for _, share := range v.PubShares {
-		pubshares = append(pubshares, sszPubkey{Pubkey: share})
-	}
-
-	// Convert to bytes ignoring errors, assume that verify has been called.
-	// TODO(corver): Extend genssz to handle errors in transform functions.
-	feeRecipient, _ := from0xHex(v.FeeRecipientAddress, 20)
-	withdrawal, _ := from0xHex(v.WithdrawalAddress, 20)
-
-	return validatorSSZ{
-		PubKey:              v.PubKey,
-		PubShares:           pubshares,
-		FeeRecipientAddress: feeRecipient,
-		WithdrawalAddress:   withdrawal,
-	}
-}
-
-type validatorJSON struct {
-	PubKey              ethHex                  `json:"public_key"`
-	PubShares           []ethHex                `json:"public_shares"`
-	FeeRecipientAddress string                  `json:"fee_recipient_address"`
-	WithdrawalAddress   string                  `json:"withdrawal_address"`
-	BuilderRegistration builderRegistrationJSON `json:"builder_registration"`
-}
-
-func validatorsToJSON(vals []Validator) []validatorJSON {
-	var resp []validatorJSON
-	for _, val := range vals {
-		var pubshares []ethHex
-		for _, share := range val.PubShares {
-			pubshares = append(pubshares, share)
-		}
-
-		resp = append(resp, validatorJSON{
-			PubKey:              val.PubKey,
-			PubShares:           pubshares,
-			FeeRecipientAddress: val.FeeRecipientAddress,
-			WithdrawalAddress:   val.WithdrawalAddress,
-			BuilderRegistration: registrationToJSON(val.BuilderRegistration),
-		})
-	}
-
-	return resp
-}
-
-func validatorsFromJSON(vals []validatorJSON) ([]Validator, error) {
-	var resp []Validator
-	for _, val := range vals {
-		var pubshares [][]byte
-		for _, share := range val.PubShares {
-			pubshares = append(pubshares, share)
-		}
-
-		reg, err := registrationFromJSON(val.BuilderRegistration)
-		if err != nil {
-			return nil, errors.Wrap(err, "validator registration from json")
-		}
-
-		resp = append(resp, Validator{
-			PubKey:              val.PubKey,
-			PubShares:           pubshares,
-			FeeRecipientAddress: val.FeeRecipientAddress,
-			WithdrawalAddress:   val.WithdrawalAddress,
-			BuilderRegistration: reg,
-		})
-	}
-
-	return resp, nil
-}
-
-type validatorSSZ struct {
-	PubKey              []byte      `ssz:"ByteList[256]"`
-	PubShares           []sszPubkey `ssz:"CompositeList[65536]"`
-	FeeRecipientAddress []byte      `ssz:"Bytes20"`
-	WithdrawalAddress   []byte      `ssz:"Bytes20"`
-}
-
-type sszPubkey struct {
-	Pubkey []byte `ssz:"ByteList[256]"`
+// ValidatorPublicKey returns the valIdx'th validator BLS group public key.
+func (c Cluster) ValidatorPublicKey(valIdx int) (tbls.PublicKey, error) {
```

Comment:
nit: `valIdx` is confusing to me as it refers to `Validator Index` on beacon chain which it is not. I suppose renaming it to `indx` would be fine